src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -145,81 +147,32 @@ public unsafe BitArray(bool[] values)
             // (true for any non-zero values, false for 0) - any values between 2-255 will be interpreted as false.
             // Instead, We compare with zeroes (== false) then negate the result to ensure compatibility.
 
-            if (Avx2.IsSupported)
+            ref byte value = ref Unsafe.As<bool, byte>(ref MemoryMarshal.GetArrayDataReference<bool>(values));
+
+            if (Vector256.IsHardwareAccelerated)
             {
-                // JIT does not support code hoisting for SIMD yet
-                Vector256<byte> zero = Vector256<byte>.Zero;
-                fixed (bool* ptr = values)
+                for (; (i + Vector256ByteCount) <= (uint)values.Length; i += Vector256ByteCount)
                 {
-                    for (; (i + Vector256ByteCount) <= (uint)values.Length; i += Vector256ByteCount)
-                    {
-                        Vector256<byte> vector = Avx.LoadVector256((byte*)ptr + i);
-                        Vector256<byte> isFalse = Avx2.CompareEqual(vector, zero);
-                        int result = Avx2.MoveMask(isFalse);
-                        m_array[i / 32u] = ~result;
-                    }
+                    Vector256<byte> vector = Vector256.LoadUnsafe(ref value, i);
+                    Vector256<byte> isFalse = Vector256.Equals(vector, Vector256<byte>.Zero);
+
+                    uint result = isFalse.ExtractMostSignificantBits();
```

Comment:
`ExtractMostSignificantBits` behaves just like `MoveMask` on x86/x64. This is also exposed by `WASM` as `bitmask`